crates/biome_js_analyze/src/lint/nursery/no_restricted_elements.rs

```diff
@@ -0,0 +1,147 @@
+use biome_analyze::context::RuleContext;
+use biome_analyze::{Ast, Rule, RuleDiagnostic, RuleSource, declare_lint_rule};
+use biome_console::markup;
+use biome_deserialize::{
+    Deserializable, DeserializableType, DeserializableValue, DeserializationContext,
+};
+use biome_js_syntax::jsx_ext::AnyJsxElement;
+use biome_rowan::AstNode;
+use rustc_hash::FxHashMap;
+use serde::{Deserialize, Serialize};
+use std::fmt::Debug;
+
+#[cfg(feature = "schemars")]
+use schemars::JsonSchema;
+
+declare_lint_rule! {
+    /// Disallow user JSX elements.
+    ///
+    /// This rule disallows the use of certain custom JSX elements as specified in the rule options.
```

Comment:
```suggestion
    /// This rule disallows the use of configured elements. Without elements configured, this rule doesn't do anything.
```
There's no reason to mention JSX because the documentation will do that for us.